evm/vm/evm.py

```diff
@@ -246,18 +252,12 @@ def import_block(self, block):
                 block.header.parent_hash, block.header.hash))
         init_header = self.create_header_from_parent(parent_header)
         parent_evm = type(self)(self.db, init_header)
-
-        # TODO: weird that this vm instance is instantiated with whatever
-        # header is currently in place and then that header is immediately
-        # discarded.
-        # Now it actually makes sense as the instantiation of the EVM for the
-        # parent's block is right above (as opposed to inside the VM's
-        # import_block method), so it becomes clear that we import the block
-        # using that EVM for the parent and once everything is validated
-        # and saved in the DB we advance this EVM to the recently-imported block.
         imported_block = parent_evm.get_vm().import_block(block)
+        self.validate_block(imported_block)
 
         persist_block_to_db(self.db, imported_block)
+        # FIXME: We must only do this when the imported block has higher
```

Comment:
Just to standardize comments, lets use `# TODO:` for easy grepping of the codebase of *things* that need to be addressed.